pkg/refactor/logengine/logger.go

```diff
@@ -0,0 +1,118 @@
+package logengine
+
+import (
+	"context"
+	"fmt"
+	"sync"
+	"time"
+
+	"github.com/direktiv/direktiv/pkg/flow/database/recipient"
+	"github.com/google/uuid"
+	"go.uber.org/zap"
+)
+
+type LogNotify interface {
+	NotifyLogs(recipientID uuid.UUID, recipientType recipient.RecipientType)
+}
+
+type Loggerw struct {
+	Sugar *zap.SugaredLogger
+	Store LogStore
+	Pub   LogNotify
```

Comment:
Why a logger needs to notify things. Logger should write only logs.

Reply:
because its possible to subscribe for new logs, its is required for streaming logs